### ISSUE TITLE
run_deployment() causing an infinite loop

### ISSUE THREAD

**Issue by devanshdoshi9** [PrefectHQ/prefect#7953]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

I was trying to run a deployment using `deployment_run()` but forgot to start the agent in another terminal. When I executed my python file it ended up getting stuck in an infinite loop. I don't think this is an intended behavior because when using the CLI to run a deployment without an agent running in the background, it still completes the execution and returns a `FlowRun` object and when the agent gets online it executes that flow.

### Reproduction

```python3
# Run the following python code without starting any agents
from prefect.deployments import run_deployment

def main():
   run_deployment(name="flow-name/deployment-name")

if __name__ == "__main__":
   main()
```

### Error

```python3
<infinite loop>
```

### Versions

```Text
Version:             2.7.3
API version:         0.8.4
Python version:      3.9.4
Git commit:          26c145b3
Built:               Fri, Dec 16, 2022 12:08 PM
OS/Arch:             win32/AMD64
Profile:             default
```

### Root cause
When we execute `deployment_run("flow-name/deployment-name", timeout=0)` we don't encounter an infinite loop and a `FlowRun` object is returned but when we run the same thing without `timeout=0` we get stuck in an infinite loop. For `deployment_run()` to work as expected either of the two cases should hold true:
1. timeout should be equal to 0 so that the below condition can be true.
```
if timeout == 0:
        return flow_run
```
2. Flow-run should be in the terminal state but since no agent is running, the flow never reaches the terminal state and ends up getting stuck in an infinite loop here.
```
with anyio.move_on_after(timeout):
        while True:
            flow_run = await client.read_flow_run(flow_run_id)
            flow_state = flow_run.state
            if flow_state and flow_state.is_final():
                return flow_run
            await anyio.sleep(poll_interval)

    return flow_run
```

### Suggestion
This error can be fixed by changing the default timeout value in `deployment_run()` from `None` to `0`. This will ensure that when `timeout` isn't set explicitly it will pick the default value of `0` and return the `FlowRun` object as per the first case discussed above. When the agent with the specified working queue starts your flow runs instantly. This will be a consistent behavior with CLI `prefect deployment run <flow-name/deployment-name>`

**Comment by madkinsz** [comment]
Hi! Thanks for the well written issue.

The default here is intentionally `None` — which means poll indefinitely. We cannot change the default behavior as it'd be a breaking change for our existing users.

**Comment by devanshdoshi9** [comment]
Agreed, the suggested solution will have breaking changes for existing users. Still, wondering why running a deployment from CLI using `prefect deployment run <flow-name/deployment-name>` doesn't cause it to poll indefinitely? Shouldn't running from CLI and python code both have similar results?

**Comment by madkinsz** [comment]
`prefect deployment run` doesn't have any polling capability yet, although we plan to add it in the future. The intended usage is different, we expect `run_deployment` to be called from within another flow and wait for completion. 

Perhaps we should have been more careful to have the default behavior match here, but we did not see these as parallel interfaces when we added the `run_deployment` utility.

We can't change it so I'm not sure if there's value to debating it though.
